### PR TITLE
Add provider='any' rule for MFA

### PR DESCRIPTION
**Pull request by santiagoaguiar**
We will support specifying `provider = 'any'` in MFA rules to let the server pick the appropriate provider based on dashboard configuration.

### CODE REVIEW

**Comment by fyockm** [comment]
@santiagoaguiar we working through the process to make rules template actual javascript (instead of markdown), and add tests. could you please add this rule to `src/rules`, and include a test for it?

File: rules/mfa.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+This rule is used to trigger multifactor authentication when a condition is met.
+
+```js
+function (user, context, callback) {
+  /*
+  You can trigger MFA conditionally by checking:
+  1. Client ID:
+  context.clientID === 'REPLACE_WITH_YOUR_CLIENT_ID'
+  2. User metadata:
+  user_metadata.use_mfa
```
Comment (pmalouin) [inline rules/mfa.md]: Minor: Maybe show that `user_metadata` is accessed like this: `user.user_metadata.use_mfa`.

**Comment by santiagoaguiar** [comment]
@fyockm Great :) Added the tests.

File: rules/mfa.md (hunk 1 above)
Comment (santiagoaguiar) [reply rules/mfa.md]: Fixed, tks!

**Pull request merged**